src/superannotate/lib/core/usecases/items.py

```diff
@@ -809,3 +814,275 @@ def execute(self):
+            if "id" in item:
+                self.item_ids.append(item["id"])
+            else:
+                if "items" not in self.path_separated[item["path"]]:
+                    self.path_separated[item["path"]]["items"] = []
+
+                self.path_separated[item["path"]]["items"].append(item)
+
+        # Removing paths that have incorrect folders in them
+        # And adding their items to "skipped list" and removing it from self.path_separated
+        # so that we don't query them later.
+        # Otherwise include folder in path object in order to later run a query
+
+        removeables = []
+        for path, value in self.path_separated.items():
+
+            project, folder = extract_project_folder(path)
+
+            if project != self.project.name:
+                removeables.append(path)
+                continue
+
+            # If no folder was provided in the path use "root"
+            # Problems with folders name 'root' are going to arise
+
+            if not folder:
+                value["folder"] = self.root_folder
+                continue
+            folder_found = False
+            try:
+                folder_candidates = SearchFoldersUseCase(
+                    project=self.project,
+                    folder_name=folder,
+                    folders=self.folder_repository,
+                    condition=Condition.get_empty_condition(),
+                ).execute()
+
+                for f in folder_candidates.data:
+                    if f.name == folder:
+                        value["folder"] = f
+                        folder_found = True
+                        break
+                # If the folder did not exist add to skipped
+                if not folder_found:
+                    removeables.append(path)
+
+            except Exception as e:
+                removeables.append(path)
+
+        ## Removing completely incorrect paths and their items
+        for item in removeables:
+            self.results["skipped"].extend(self.path_separated[item]["items"])
+            self.path_separated.pop(item)
+
+    def __build_query_string(self, path, item_names):
+        _, folder = extract_project_folder(path)
+        if not folder:
+            folder = "root"
+        query_str = f"metadata(name IN {str(item_names)}) AND folder={folder}"
+
+        return query_str
+
+    def __query(self, path, items):
+        folder = None
+        _, folder = extract_project_folder(path)
+
+        item_names = [item["name"] for item in items["items"]]
+        query = self.__build_query_string(path, item_names)
+        query_use_case = QueryEntitiesUseCase(
+            reporter=self.reporter,
+            project=self.project,
+            backend_service_provider=self._backend_client,
+            query=query,
+            folder=items["folder"],
+            subset=None,
+        )
+
+        try:
+            queried_items = query_use_case.execute().data
```

Comment:
no exception should be thrown there
you can check if there is any error via UseCase.execute().errors